lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeDynamicTypeResolution.cpp

```diff
@@ -3034,6 +3034,116 @@ bool SwiftLanguageRuntimeImpl::IsStoredInlineInBuffer(CompilerType type) {
   return true;
 }
 
+CompilerType SwiftLanguageRuntimeImpl::ResolveTypeAlias(CompilerType alias) {
+  using namespace swift::Demangle;
+  Demangler dem;
+
+  auto tss = alias.GetTypeSystem().dyn_cast_or_null<TypeSystemSwift>();
+  if (!tss)
+    return {};
+  auto tr_ts = tss->GetTypeSystemSwiftTypeRef();
+  if (!tr_ts)
+    return {};
+
+  // Extract the mangling of the alias type's parent type and its
+  // generic substitutions if any.
+  auto mangled = alias.GetMangledTypeName().GetStringRef();
+  NodePointer demangled = dem.demangleSymbol(mangled);
+  NodePointer type = swift_demangle::GetType(demangled);
+  if (!type || type->getKind() != Node::Kind::TypeAlias ||
+      type->getNumChildren() != 2)
+    return {};
```

Comment:
Should we add some logs if there is a type but not in the structure we expect? There might be some valid demangle trees that we don't know about.